turbopack/crates/turbo-tasks-backend/src/lmdb_backing_storage.rs

```diff
@@ -411,25 +469,36 @@ impl BackingStorage for LmdbBackingStorage {
             tx.commit()
                 .with_context(|| anyhow!("Unable to commit operations"))?;
         }
+        {
+            let _span = tracing::trace_span!("swap read transactions").entered();
+            // This resets the thread local storage for read transactions, read transactions are
+            // eventually dropped, allowing DB to free up unused storage.
+            self.read_transactions_cache
+                .store(Arc::new(ThreadLocal::new()));
+        }
         span.record("db_operation_count", op_count);
         Ok(())
     }
 
     fn start_read_transaction(&self) -> Option<ReadTransaction> {
-        Some(Self::from_tx(self.env.begin_ro_txn().ok()?))
+        Some(Self::from_tx(self.begin_read_transaction().ok()?))
     }
 
     unsafe fn end_read_transaction(&self, transaction: ReadTransaction) {
-        ManuallyDrop::into_inner(self.to_tx(transaction))
-            .commit()
-            .unwrap();
+        self.end_read_transaction(ManuallyDrop::into_inner(Self::to_tx(self, transaction)));
     }
 
     unsafe fn forward_lookup_task_cache(
         &self,
         tx: Option<ReadTransaction>,
         task_type: &CachedTaskType,
     ) -> Option<TaskId> {
+        // Performance optimization when the database was empty
+        // It's assumed that no cache entries are removed from the memory cache, but we might change
+        // that in future.
```

Comment:
:+1: Seems like a nice optimization for cold starts.

We can make this an `AtomicBool` and dynamically flip this to false when we start doing memory cache eviction.

If lmdb doesn't have a filter already, maybe we generalize this optimization by adding our own probabilistic filter (bloom filter or whatever else) in front to avoid attempting reads that won't be in the cache?